### PR TITLE
[aes] Rework SecAllowForcingMasks parameter

### PR DESCRIPTION
**Pull request by vogelpi**
This PR reworks the `SecAllowForcingMasks` parameter of AES to enable it for ES. The PR is big because it touches RTL, DV, DIF, Software. But actual changes are simple:
 - Instead of forcing the PRNG output to zero (not suitable for ES), the output is just forced to be constant/not further advance and keep the current state.
 - The configuration bit is renamed from FORCE_ZERO_MASKS to FORCE_MASKS.
 - The configuration bit is moved from the shadowed main control register to the shadowed auxiliary control register (also has REGWEN).

This is related to lowRISC/OpenTitan#14240.

### CODE REVIEW

File: sw/device/lib/dif/dif_aes_unittest.cc [hunk 1]
```diff
@@ -634,15 +614,17 @@ TEST_F(RessedOnKeyChangeTest, Unlock) {
   });
 
   transaction_.reseed_on_key_change = true;
-  transaction_.reseed_on_key_change_lock = false;
+  transaction_.force_masks = true;
+  transaction_.ctrl_aux_lock = false;
```
Comment (engdoreis) [inline sw/device/lib/dif/dif_aes_unittest.cc:618]: Nit. This is he default value already, so I would drop this line.

File: sw/device/lib/dif/dif_aes_unittest.cc [hunk 2]
```diff
@@ -652,13 +634,14 @@ TEST_F(RessedOnKeyChangeTest, Lock) {
   });
 
   transaction_.reseed_on_key_change = true;
-  transaction_.reseed_on_key_change_lock = true;
+  transaction_.force_masks = false;
```
Comment (engdoreis) [inline sw/device/lib/dif/dif_aes_unittest.cc:637]: Nit. This is he default value already, so I would drop this line.

File: sw/device/lib/dif/dif_aes_unittest.cc [hunk 3]
```diff
@@ -668,17 +651,41 @@ TEST_F(RessedOnKeyChangeTest, LockedSuccess) {
...
```
Comment (engdoreis) [inline sw/device/lib/dif/dif_aes_unittest.cc:654]: Nit. This is he default value already, so I would drop this line.

File: sw/device/lib/dif/dif_aes_unittest.cc [hunk 4]
```diff
@@ -668,17 +651,41 @@ TEST_F(RessedOnKeyChangeTest, LockedSuccess) {
...
+  EXPECT_READ32(AES_STATUS_REG_OFFSET, 1);
+  ExpectConfig({
+      .key_len = AES_CTRL_SHADOWED_KEY_LEN_VALUE_AES_128,
+      .mode = AES_CTRL_SHADOWED_MODE_VALUE_AES_ECB,
+      .operation = AES_CTRL_SHADOWED_OPERATION_VALUE_AES_ENC,
+      .mask_reseed = AES_CTRL_SHADOWED_PRNG_RESEED_RATE_VALUE_PER_1,
+  });
+
+  transaction_.reseed_on_key_change = true;
+  transaction_.force_masks = false;
```
Comment (engdoreis) [inline sw/device/lib/dif/dif_aes_unittest.cc:676]: Nit. This is he default value already, so I would drop this line.

File: sw/device/lib/dif/dif_aes_unittest.cc [hunk 5]
```diff
@@ -688,12 +695,14 @@ TEST_F(RessedOnKeyChangeTest, LockedError) {
   });
 
   transaction_.reseed_on_key_change = true;
-  transaction_.reseed_on_key_change_lock = true;
+  transaction_.force_masks = false;
```
Comment (engdoreis) [inline sw/device/lib/dif/dif_aes_unittest.cc:698]: Nit. This is he default value already, so I would drop this line.

**Review by engdoreis — APPROVED** [review]
Thanks @vogelpi for the PR. The SW part LGTM.

File: sw/device/sca/aes_serial.c [hunk 6]
```diff
@@ -449,7 +449,7 @@ bool test_main(void) {
   init_aes();
 
 #if !OT_IS_ENGLISH_BREAKFAST
-  if (transaction.masking == kDifAesMaskingForceZero) {
+  if (transaction.force_masks == true) {
```
[the remark this answers is not in the document]
Comment (vogelpi) [reply sw/device/sca/aes_serial.c]: Done.

File: sw/device/lib/dif/dif_aes_unittest.cc (hunk 2 above)
Comment (vogelpi) [reply sw/device/lib/dif/dif_aes_unittest.cc]: Sounds good, done here and for all the other cases below.

**Comment by vogelpi** [comment]
> Thanks @vogelpi for the PR. The SW part LGTM.

Many thanks for your review @engdoreis ! I've implemented your suggestions. If someone could review the hardware part this would be great. The changes are small and inline with what we agree in #14240 .

**Review by andreaskurth — APPROVED** [review]
I'm not an expert on this code, but what I understood LGTM. Thanks @vogelpi!

**Comment by vogelpi** [comment]
Thanks for your review @andreaskurth ! I am merging this to move forward. If anything else should come up, I am happy to do a follow-up PR.

**Review by cdgori — COMMENTED** [review]
Much delayed but I also had a quick look at all the hardware changes and LGTM!

**Comment by vogelpi** [comment]
Thanks @cdgori for taking a look! :+1:

**Pull request merged**